spec/editors_draft.html

```diff
@@ -3298,6 +3298,231 @@ <h2>Evaluation Semantics</h2>
+      <pre id="load-renaming-blank-example-consistent-ids"
+         title="Producing consistent blank node identifiers shared between RDF and Composite Values"
+         data-transform="updateExample"
+         data-content-type="text/turtle"
+         class="nohighlight example">
+      <!--
+      _:b0 ex:p3 "[ _:b0 ]"^^cdt:List .
+      -->
+      </pre>
+      <pre id="load-renaming-blank-example-consistent-ids-result"
+         title="Producing consistent blank node identifiers shared between RDF and Composite Values: Result"
+         data-transform="updateExample"
+         data-content-type="text/n-triples"
+         class="nohighlight example">
+      <!--
+      _:b004 <http://example.org/p3> "[ _:b004 ]"^^<http://w3id.org/awslabs/neptune/SPARQL-CDTs/List> .
+      -->
+      </pre>
+
+      <p>Loading the Turtle in <a href="#load-renaming-blank-reuse"></a> with a triple with a <a>cdt:Map literal</a> containing two map entries, both having the same blank node value, MUST result in a dataset with a single triple whose <a>cdt:Map literal</a> object contains two map entries having the same blank node value, and whose identifier MUST NOT be "b0" (<a href="#load-renaming-blank-reuse-result"></a>):</p>
+      <pre id="load-renaming-blank-reuse"
+         title="Producing consistent blank node identifiers used multiple times in Composite Values"
+         data-transform="updateExample"
+         data-content-type="text/turtle"
+         class="nohighlight example">
+      <!--
+      ex:s ex:p3 "{ 'this': _:b0, 'again': _:b0 }"^^cdt:Map .
+      -->
+      </pre>
+      <pre id="load-renaming-blank-reuse-result"
+         title="Producing consistent blank node identifiers used multiple times in Composite Values: Result"
+         data-transform="updateExample"
+         data-content-type="text/n-triples"
+         class="nohighlight example">
+      <!--
+      <http://example.org/s> <http://example.org/p3> "{'this': _:b005,'again': _:b005}"^^<http://w3id.org/awslabs/neptune/SPARQL-CDTs/Map> .
+      -->
+      </pre>
+    </section>
+
+    <section>
+      <h2>Exporting Requirements</h2>
+      <p>
+        Conforming implementations MUST process <a>cdt:List literal</a>s and <a>cdt:Map literal</a>s during export, replacing the substrings containing the serialized form of any blank nodes with identifier <var>id</var> with a blank node label <var>b</var>, such that
+      </p>
+
+      <ul>
+        <li>within the scope of the document being serialized, all occurrences of <var>id</var> as a blank node identifier are replaced with the same <var>b</var> value (both inside and outside of composite values), and</li>
+        <li>within the scope of the document being serialized, two distinct <var>id</var> values are replaced with distinct <var>b</var> values.</li>
+      </ul>
+
+      <p>
+        Here, "exporting" is any single process by which composite values leave a SPARQL system or RDF Dataset. This includes (but is not limited to):
```

Comment:
Also here:
* "single"?
* "RDF Dataset"?

Reply:
"Single" here is meant to mirror the language used in the importing section (which I think has a separate PR comment from you which I will respond to).

I'm not entirely sure what your issue with "RDF Dataset" is. I mean [this](https://www.w3.org/TR/sparql11-query/#rdfDataset). The idea is that you have data in a dataset upon which the query is being evaluated (whether that's in memory or in some persistent storage), and the critical point here is when data leaves that storage system and  is returned to the user/client.

Reply:
How's about the following?

"...  by which a SPARQL system writes data in a form to be consumed by external processes in other systems."

"writes" may be replaced by "serializes"

Reply:
> Thanks for the clarification regarding "RDF Dataset". I guess I find it somehow inconsistent to mention a SPARQL system and an RDF dataset together here, because these two are very different types of things. Also, where would such an RDF dataset be, if not in a SPARQL system?

Well, I could imagine different services sitting on top of a dataset. In 1G, we might have both a SPARQL system and an OpenCypher system. And while it is also part of the suite of SPARQL specs, the graph store protocol is a different type of service that provides a more direct interaction with the dataset than SPARQL Query/Update. I can remove the language about datasets and talk just about "SPARQL systems" if you feel strongly about this, but I do think the extra language adds something here.